Student-Unity/src/main/java/csed/swe/studentunity/dao/AnnouncementRepo.java

```diff
@@ -15,4 +17,15 @@ public interface AnnouncementRepo extends JpaRepository<Announcement, UUID> {
                     "JOIN course AS c ON rc.course_id = c.course_id " +
             "WHERE rc.id = ?1", nativeQuery = true)
     Optional<Iterable<Object>> getAnnouncementByRegisteredCourse(Long userId);
+
+    @Query(value =
+            "SELECT course_name, body, posted_date " +
+            "FROM announcement AS a JOIN registered_course AS rc ON a.course_id = rc.course_id " +
+                    "JOIN course AS c ON rc.course_id = c.course_id " +
+            "WHERE rc.id = ?1 AND a.course_id = ?2", nativeQuery = true)
+    Optional<List<Object>> filterAnnouncements(Long userId, Long courseId);
+
+    @Modifying
+    @Query( value = "DELETE FROM announcement WHERE id = ?1", nativeQuery = true)
+    void deleteById(Long announcementId);
```

Comment:
i think if you typed ``deleteAnnouncementById(Long announcementId);``, it should also work.

Reply:
I like to make the Queries natively And not rely on JPA auto-generated queries